test/clientTest.js

```diff
@@ -237,20 +234,15 @@ mc.supportedVersions.forEach(function (supportedVersion, i) {
         })
       })
 
-      it('gets kicked when no credentials supplied in online mode', function (done) {
+      it('Gets kicked when no credentials supplied in online mode', function (done) {
         const client = mc.createClient({
           username: 'Player',
           version: version.minecraftVersion,
           port: PORT
         })
         client.on('error', err => done(err))
-        let gotKicked = false
         client.on('disconnect', function (packet) {
           assert.ok(packet.reason.indexOf('"Failed to verify username!"') !== -1 || packet.reason.indexOf('multiplayer.disconnect.unverified_username') !== -1)
-          gotKicked = true
```

Comment:
Why ?

Reply:
There's no reason to wait for the bot to end, therefore the `gotKicked` variable is not needed.

Reply:
???
The purpose of the test is to make sure we get kicked and exit properly, so yes this is required.

Reply:
Not if you only finish the test when you get disconnected and the reason is that you got kicked. Otherwise send an error. Make things simple

Reply:
"disconnect" event isn't "you got disconnected", it's a packet.
We received that packet, mark that we got kicked, and then when the client end make sure we indeed for that packet before getting disconnected